src/client/systems/NetworkingClient.cpp

```diff
@@ -15,150 +15,218 @@
-    return mPlayerId.get();
+/* Main thread methods */
+
+void NetworkingClient::createTcpSocket(unsigned short port) {
+    std::lock_guard<std::mutex> mTcpSocket_guard(mTcpSocket_lock);
+    mTcpSocket_t = std::unique_ptr<sf::TcpSocket>(new sf::TcpSocket);
+    mTcpSocket_t->connect(SERVER_IP, port);
 }
 
-void NetworkingClient::setClientUnreliableUpdate(ClientUnreliableUpdate client_unreliable_update) {
-    mClientUnreliableUpdate.set(client_unreliable_update);
+void NetworkingClient::createUdpSocket() {
+    std::lock_guard<std::mutex> mUdpSocket_guard(mUdpSocket_lock);
+    mUdpSocket_t = std::unique_ptr<sf::UdpSocket>(new sf::UdpSocket);
+    mUdpSocket_t->bind(0);
+    mUdpSocket_t->setBlocking(false);
 }
 
-void NetworkingClient::setClientReliableUpdate(ClientReliableUpdate client_reliable_update) {
-    mClientReliableUpdate.set(client_reliable_update);
+bool NetworkingClient::registerNetworkingClient() {
+    sf::Packet registration_request;
+    if (registration_request << ReliableCommandType::register_client) {
+        std::lock_guard<std::mutex> mTcpSocket_guard(mTcpSocket_lock);
+        mTcpSocket_t->send(registration_request);
+        return readRegistrationResponse();
+    } else {
+        std::cout << "Failed to form packet" << std::endl;
+        return false;
+    }
 }
 
-void NetworkingClient::readRegistrationResponse() {
-    // read registration data
+bool NetworkingClient::readRegistrationResponse() {
     sf::Packet registration_response;
     ReliableCommand reliable_command;
 
-    if (mReliableClient->receive(registration_response) == sf::Socket::Done) {
+    if (mTcpSocket_t->receive(registration_response) == sf::Socket::Done) {
         if (registration_response >> reliable_command &&
             reliable_command.command == ReliableCommandType::register_client) {
-            std::cout
-                << "Registered with ID and current tick: "
-                << reliable_command.client_id
-                << " "
-                << reliable_command.tick
-                << std::endl;
-            mClientId = reliable_command.client_id;
-
-            mTick = reliable_command.tick;
-
-            mIsRegistered = true;
+            mClientId_ta = static_cast<uint>(reliable_command.client_id);
+            mTick_ta = static_cast<uint>(reliable_command.tick);
+            return true;
         }
     }
+    return false;
 }
 
-void NetworkingClient::registerNetworkingClient() {
-    sf::Packet registration_request;
-    if (registration_request << ReliableCommandType::register_client) {
-        mReliableClient->send(registration_request);
-        readRegistrationResponse();
-    }
+int NetworkingClient::getPlayerId() const {
+    return mPlayerId_ta;
+}
 
-    if (!mIsRegistered) {
-        throw std::runtime_error("Failed to register. Exiting.");
-    }
+GameState NetworkingClient::getGameState() {
+     std::lock_guard<std::mutex> mGameState_guard(mGameState_lock);
+    mGameStateIsFresh_ta = false;
```

Comment:
what is _ta?

Reply:
I added a description in the README.txt about this.